outline/tun2socks/tunnel_android.go

```diff
@@ -40,7 +40,7 @@ func init() {
 //
 // Returns an error if the TUN file descriptor cannot be opened, or if the tunnel fails to
 // connect.
-func ConnectShadowsocksTunnel(fd int, client *shadowsocks.Client, isUDPEnabled bool) (Tunnel, error) {
+func ConnectShadowsocksTunnel(fd int, client *shadowsocks.Client, isUDPEnabled bool) (OutlineTunnel, error) {
```

Comment:
Does this break backwards compatibility? If we are breaking it, then keeping the legacy api doesn't help much.

Reply:
Changing from `Tunnel` to `OutlineTunnel` will keep backward compatibility, this is the old JNI interface (it requires `OutlineTunnel` instead of `Tunnel`):

<img width="847" alt="image" src="https://user-images.githubusercontent.com/93548144/234164262-664cd33a-181f-48a2-8892-9a998298ea92.png">